WcaOnRails/app/models/competition.rb

```diff
@@ -261,4 +285,25 @@ def end_date
       errors.add(:eventSpecs, "invalid event ids: #{invalid_events.map(&:id).join(',')}")
     end
   end
+
+  def competitions_nearby
+    Competition.where(
+      "ABS(DATEDIFF(?, CONCAT(year, '-', month, '-', day))) <= ? AND id <> ?", start_date, Competition.NEARBY_DAYS_WARNING, id)
+      .select { |c| kilometers_to(c) <= Competition.NEARBY_DISTANCE_KM_WARNING }
+      .sort_by { |c| kilometers_to(c) }
+  end
+
+  # Source http://www.movable-type.co.uk/scripts/latlong.html
+  def kilometers_to(c)
+    def to_radians(degrees)
```

Comment:
FYI: Ruby doesn't have nested methods (http://stackoverflow.com/a/4865161). So `to_radians` would be an instance method on `Competition`.

Anyway, I think the method is fine within the `Competition` class, just make it private and reuse it like so:

```ruby
def latitude_radians
  to_radians latitude_degrees
end

def longitude_radians
  to_radians longitude_degrees
end

# ...
private def to_radians(degrees)
  degrees * Math::PI / 180
end
```